internal/mining/cpuminer/cpuminer.go

```diff
@@ -39,16 +38,19 @@ const (
 	// keep track of the hashes per second.
 	hashUpdateSecs = 15
 
-	// maxSimnetToMine is the maximum number of blocks to mine on HEAD~1
-	// for simnet so that you don't run out of memory if tickets for
-	// some reason run out during simulations.
+	// maxSimnetToMine is the maximum number of blocks mined on HEAD~1 for
+	// simnet that fail to submit to avoid pointlessly mining blocks in
+	// situations such as tickets running out during simulations.
 	maxSimnetToMine uint8 = 4
 )
 
 var (
-	// defaultNumWorkers is the default number of workers to use for mining
-	// and is based on the number of processor cores.  This helps ensure the
+	// MaxNumWorkers is the maximum number of workers that will be allowed for
+	// mining and is based on the number of processor cores.  This helps ensure
 	// system stays reasonably responsive under heavy load.
+	MaxNumWorkers = uint32(runtime.NumCPU() * 2)
```

Comment:
I dislike this and wonder if this isn't actually slower. We should cap use at all cores IMO.

Reply:
I tested it on a Zen and with SMT enabled it is definitely faster to allow it to use the additional threads.

That said, it doesn't use that many workers by default and only imposes a max limit for the number the user can request to use.

Also, since the CPU miner is only used for testnet, regnet, simnet, it is intentionally disabled for mainnet, the default is 1 worker (per the constant below) because the low difficulties involved means running multiple workers just ends up producing a heap of orphans anyway.